analyzer/tools/build-logger/tests/unit/__init__.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python3
+# coding=utf-8
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+import json
+import os
+import shlex
+import subprocess
+import tempfile
+import unittest
+from typing import Mapping, Optional, Tuple, Sequence
+
+REPO_ROOT = os.path.abspath(os.environ["REPO_ROOT"])
+LOGGER_DIR = os.path.join(REPO_ROOT, "build")
+
+
+class quote:
+    char = '"'
+
+
+class backslash:
+    char = "\\"
+
+
+class bell:
+    char = "\a"
+
+
+class backspace:
+    char = "\b"
+
+
+class tab:
+    char = "\t"
+
+
+class newline:
+    char = "\n"
+
+
+class vericaltab:
+    char = "\v"
+
+
+class carrigereturn:
+    char = "\r"
+
+
+# The following 3 characters are below 0x20, so they need special treat.
+# However, these are not *explicitly* escaped by the ldlogger.
+class ack:
+    char = "\x06"
+
+
+class shiftin:
+    char = "\x0F"
+
+
+class datalinkescape:
+    char = "\x10"
+
+
+class escape:
+    char = "\x1B"
+
+
+class formfeed:
+    char = "\f"
+
+
+def concat(*args):
+    res = ""
+    for arg in args:
+        ntimes = 1
+        if type(arg) == tuple and len(arg) == 2:
+            ntimes = arg[1]
+            arg = arg[0]
+        char = getattr(arg, "char", None)
+        if type(arg) == str:
+            res += arg
+            continue
+        if not char:
+            raise KeyError()
+        res += char * ntimes
+    return res
+
+
+def _run_command(
+    cmd: Sequence[str],
+    cwd: Optional[str] = None,
+    env: Optional[Mapping[str, str]] = None,
+) -> Tuple[int, str, str]:
+    cmd = " ".join([shlex.quote(c) for c in cmd])
+    try:
+        proc = subprocess.Popen(
+            cmd,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            cwd=cwd,
+            env=env,
+            shell=True,
+            encoding="utf-8",
+            universal_newlines=True,
+            errors="ignore",
+        )
+        stdout, stderr = proc.communicate()
+        retcode = proc.returncode
+    except FileNotFoundError:
+        return 2, "", ""
+    return retcode, stdout, stderr
+
+
+_PRINT_STRINGIFIED_VARIABLE_CODE = """
+#include <stdio.h>
+#define xstr(a) str(a)
+#define str(a) #a
+int main() {
+  char buf[] = xstr(VARIABLE); // It might containt null characters.
```

Comment:
I think, It would be worth to comment that `VARIABLE` is going to be fed by the different test cases.